### DISCUSSION TITLE
Search Weight

### DISCUSSION THREAD

**Discussion by craigmoscardini** [netbox-community/netbox#17543] [General]
Is there any way to weight the search results to return in a certain order? Currently there appears to be a default device, interface, site etc, but I would like to customise that order to ensure results of a certain type appear at the top.

I see custom fields have a search weight so basically the same for built-in fields is what I am looking for

**Comment by jeremystretch** [comment]
Search results are ordered by the weight assigned to the matching field. For example, a match on a device name has a stronger significance (100) than a match on an interface description (500). So the order of object types depends on the specific results.

It would not make sense to order results by object type, because certain fields carry more significance than others.

**Comment by craigmoscardini** [comment]
OK. I understand the principle, but my question was about customising it for what I want not what you think users want? Where are those weights defined for bult-in objects? Custom fields are defined through the GUI. Built-in are not, but I am quite happy to adjust wherever they are in the code

**Comment by austinthompson00** [comment]
I think this is a great idea that would be very beneficial. Being able to add search weights to pretty much any field would be extremely helpful. Taking this as far as being able to set weights for commonly used search terms would be fantastic. I sometimes get search results with the thing I'm looking for ending up the bottom of the page, even if what I'm searching for has an exact match with the name (I'm assuming because of the object type).
